File: src/main/java/org/pl/interpreter/visitor/BoolVisitor.java
package org.pl.interpreter.visitor;

import org.pl.interpreter.IInterpreter;
import org.pl.interpreter.exception.VisitorException;
import org.pl.parser.ast.BoolNode;
import org.pl.parser.ast.INode;


/**
 * Interprets the bool node.
 */
public class BoolVisitor implements IVisitor {

    @Override
    public boolean matches(INode node) {
        return BoolNode.class == node.getClass();
    }

    @Override
    public Boolean visit(IInterpreter interpreter, INode node) throws VisitorException {
        return ((BoolNode) node).getValue();
    }
}

File: src/main/kotlin/org/leaflang/interpreter/visitor/BoolVisitor.kt
package org.leaflang.interpreter.visitor

import org.leaflang.interpreter.IInterpreter
import org.leaflang.interpreter.result.boolResult
import org.leaflang.parser.ast.INode
import org.leaflang.parser.ast.value.BoolNode

/**
 * Interprets the bool node.
 */
class BoolVisitor : IVisitor {
    override fun visit(interpreter: IInterpreter, node: INode) = boolResult((node as BoolNode).value)
}